lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -277,4 +278,29 @@ protected function prepareMessage()
 
         return $this;
     }
+
+    /**
+     * Add a file joined to the email
+     *
+     * @param string $directory is the directory the file is located
+     * @param string $fileName is the name of the file to join
+     * @param string $fileType is the type of the file, eg: application/pdf
+     * @param string $joinedFileName is the name given to the joined file
+     * @throws \Magento\Framework\Exception\FileSystemException
+     */
+    public function joinFile($directory, $fileName, $fileType, $joinedFileName)
```

Comment:
I think would be better to:
1. Rename method to attackFile
2. Combine directory & filename to one parameter - filepath
3. Make auto-filling of file type (or define some default value)
4. Make auto-filling of attacked file (by default should be the same)